courses/views.py

```diff
@@ -0,0 +1,51 @@
+from django.shortcuts import render, get_object_or_404, redirect
+from django.forms.extras.widgets import SelectDateWidget
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+from django.views.generic.edit import CreateView, UpdateView, DeleteView
+from django.forms.models import modelform_factory
+from courses.models import Course
+from courses.forms import CourseModelForm
+# Create your views here.
+class CourseDetailView(DetailView):
+    model = Course
+    template_name = "courses/course.html"
+
+class CourseListView(ListView):
+    model = Course
+    template_name = "courses/list.html"
+    context_object_name = "courses"
+
+class CourseCreateView(CreateView):
+    model = Course
+    success_url = "/courses/"
+    template_name = "courses/create.html"
+    form_class =  modelform_factory(Course,
+        widgets={"begin_date": SelectDateWidget, "end_date": SelectDateWidget })
+    def form_valid(self, form):
+        cor = form.save()
+        logger.debug("Create a Course " + cor.name)
+        logger.warning("Create a Course " + cor.name)
+        logger.info("Create a Course " + cor.name)
+        logger.error("Create a Course " + cor.name)
+        return super(CourseUpdate, self).form_valid(form)
+
+class CourseUpdateView(UpdateView):
+    model = Course
+    success_url = "/courses/"
+    template_name = "courses/edit.html"
+    form_class =  modelform_factory(Course,
+        widgets={"begin_date": SelectDateWidget, "end_date": SelectDateWidget })
+    def form_valid(self, form):
+        cor = form.save()
+        logger.debug("Change a Course " + cor.name)
```

Comment:
NameError at /courses/edit/1/
global name 'logger' is not defined
Request Method:	POST
Request URL:	http://127.0.0.1:8000/courses/edit/1/
Django Version:	1.7.1
Exception Type:	NameError
Exception Value:	
global name 'logger' is not defined
Exception Location:	/home/wowkalucky/lrn/pybursa/homeworks/m_smirnitskaya/hw11/pybursa/courses/views.py in form_valid, line 41
Python Executable:	/home/wowkalucky/.virtualenvs/PB/bin/python
Python Version:	2.7.6